File: cloud-collect-core/src/main/java/com/windea/demo/cloudcollect/core/domain/enums/ResponseResult.java
package com.windea.demo.cloudcollect.core.domain.enums;

/**
 * 响应结果。
 */
public enum ResponseResult {
	SUCCESS(200, "操作成功"),
	VALIDATION_ERROR(400, "参数错误"),
	NOT_LOGIN(401, "未登录"),
	PERMISSION_ERROR(403, "权限错误"),
	NOT_FOUND(404, "内容未找到"),
	INTERNAL_ERROR(500, "内部错误"),
	NOT_IMPLEMENTED(501, "功能未实现"),

	USER_DUPLICATE(4001, "用户已存在"),
	COLLECT_DUPLICATE(4002, "收藏已存在");


	private final int code;
	private final String message;

	ResponseResult(int code, String message) {
		this.code = code;
		this.message = message;
	}


	public int code() {
		return code;
	}

	public String message() {
		return message;
	}

	@Override
	public String toString() {
		return this.code + " " + this.message;
	}
}

File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/domain/enums/ResponseResult.kt
package com.windea.demo.cloudcollect.core.domain.enums

/**响应结果。*/
enum class ResponseResult(
	val code: Int,
	val message: String
) {
	OK(200, "200 操作成功。"),
	USER_NOT_FOUND(401, "401 用户未找到！"),
	INVALID_USER(403, "403 非法的用户！"),
	NOT_FOUND(404, "404 页面未找到！"),
	NOT_IMPLEMENTED(501, "501 功能未实现！"),
	VALIDATION_ERROR(400, "400 参数错误！")
}
